src/api/ledger/parse/pathfind.js

```diff
@@ -8,17 +8,25 @@ function parsePaths(paths) {
     _.omit(step, ['type', 'type_hex'])));
 }
 
+function ensureIOUCounterparty(amount, sourceAmount) {
+  const defaults = sourceAmount.counterparty ?
+    {counterparty: sourceAmount.counterparty} : {};
+  return amount.currency === 'XRP' || amount.counterparty ? amount :
+    _.assign(defaults, amount);
+}
+
 function parsePathfind(sourceAddress: string,
     destinationAmount: Object, pathfindResult: Object): Object {
   return pathfindResult.alternatives.map(function(alternative) {
+    const sourceAmount = parseAmount(alternative.source_amount);
     return {
       source: {
         address: sourceAddress,
-        amount: parseAmount(alternative.source_amount)
+        amount: sourceAmount
       },
       destination: {
         address: pathfindResult.destination_account,
-        amount: destinationAmount
+        amount: ensureIOUCounterparty(destinationAmount, sourceAmount)
```

Comment:
* A Payment's paths is built from Amount/SendMax/Paths

* A Payment can be cross currency/issuer, executing offers

* issuer of self(destination|source) is code for any trustline
  * this is meaningful, and allows sending from multiple source trustlines to 
    multiple destination trustlines (see: [payment-iou-multipath.json](https://github.com/ripple/ripple-lib-extensions/blob/master/transactionparser/test/fixtures/payment-iou-multipath.json))
 

Assuming the above points were true (ask around/investigate I guess), would you find simply omitting the counterparty a better representation for the semantics ?